assignments/sa/redux/index.md

```diff
@@ -379,16 +379,286 @@ git tag v4
+    super(props);
+
+    this.state = {
+      videos: [],
+      selectedVideo: null,
+    };
+
+    this.search = debounce(this.search, 300);
+    this.search('pixar');
+  }
+
+  search = (text) => {
+    youtubeSearch(text).then((videos) => {
+      this.setState({
+        videos,
+        selectedVideo: videos[0],
+      });
+    });
+  }
+
+  render() {
+    return (
+      <div>
+        <SearchBar onSearchChange={this.search} />
+        <div id="video-section">
+          <VideoDetail video={this.state.selectedVideo} />
+          <VideoList onVideoSelect={(selectedVideo) => this.setState({ selectedVideo })} videos={this.state.videos} />
+        </div>
+      </div>
+    );
+  }
+}
+```
+</details>
+
+🚀 To finish off our merge, let's create a new endpoint in `components/app.js` for YouTube:
+
+```javascript
+// Add this import somewhere at the top
+import YouTube from './youtube';
+
+// Add a new NavLink to your Nav constant
+<li><NavLink to="/youtube" exact>YouTube</NavLink></li>
+
+// And add a new Route to the Switch in App
+<Route exact path="/youtube" component={YouTube} />
+```
+
+🚀 Once you have finished handling the conflicts, go ahead and add, commit, and push the changes.
+
+To update our code, we will follow largely the same process as above, but with more complicated functionality.
+
+### Actions
+
+Since selecting a video is a new action handled by our redux store, we will need to create a new function in `actions/index.js` to go along with it.
+
+🚀 First let's expand our `ActionTypes`:
+
+```javascript
+// keys for actiontypes
+export const ActionTypes = {
+  INCREMENT: 'INCREMENT',
+  DECREMENT: 'DECREMENT',
+  SELECT: 'SELECT',
+};
+```
+
+🚀 Next, we add a new function to use our SELECT type. Notice that this function, unlike before, is passed a variable to be used as a payload.
+
+```javascript
+export function selectVideo(video) {
+  return {
+    type: ActionTypes.SELECT,
+    payload: video,
+  };
+}
+```
+
+### Reducers
+🚀 Create `reducers/select_reducer.js`
+
+```javascript
+import { ActionTypes } from '../actions';
+
+const SelectReducer = (state = null, action) => {
+  switch (action.type) {
+    case ActionTypes.SELECT:
+      return action.payload;
+    default:
+      return state;
+  }
+};
+
+export default SelectReducer;
+```
+
+It is nearly the same as our previous reducer with two key differences:
+
+1. Our initial state is now `null` to reflect the new type.
+2. We update the state by returning `action.payload` instead of manipulating the state directly. Remember that we defined the payload as `video` in our actions file.
+
+🚀 Add our new reducer to `reducers/index.js` so that the store knows to keep track of an additional state variable.
+
+```javascript
+// the starting point for your redux store
+// this defines what your store state will look like
+import { combineReducers } from 'redux';
+import CountReducer from './count-reducer';
+import SelectReducer from './select_reducer';
```

Comment:
same here as the above comment about the action types. do they need the `CountReducer`?